src/MagentoHackathon/Composer/Magento/Parser/PackageXmlParser.php

```diff
@@ -64,6 +64,8 @@ protected function parseMappings()
                     foreach ($target->children() as $child) {
                         foreach ($this->getElementPaths($child) as $elementPath) {
                             $relativePath = $basePath . '/' . $elementPath;
+                            //remove the any trailing './' or '.' from the targets base-path.
+                            $relativePath = preg_replace('#^[\/\.]*#', '', $relativePath);
```

Comment:
You've got the same preg_replace in three different places. We should try putting this in a separate method.